iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser.yaml;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SpringConfigYamlPreprocessor {
+  private static final String DOCUMENT_BREAK = "---";
+  private static final int DOCUMENT_BREAK_LENGTH = DOCUMENT_BREAK.length();
+
+  public String preprocess(String source) {
+    return Stream.of(source.split("\n", -1))
+      .map(SpringConfigYamlPreprocessor::removeInlineCommentAfterDocumentBreak)
+      .collect(Collectors.joining("\n"));
```

Comment:
Performance wise, this is should be reconsidered, as every file needs to go through here.
I would suggest to not push the array into a stream and simply operate on the array of `source.split("\n", -1)` in a for loop.
Than you can either push the resulting string into a `StringBuilder`.

Reply:
Hm, I'm not sure if it will bring any improvement. In both approaches, we would need to split the file and iterate over all the lines, and we can't know beforehand if it will be required or if the file can be skipped. I don't expect a significant difference between a stream and a string builder in this case.

Reply:
But in the case that's implemented right now, every file (regardless if there is a comment after a `---`) would be split and then joined, which is quite expensive. The StringBuilder would reduce some of the load IMO

Reply:
As discussed, I added a check that some lines need to be processed.

Reply:
Fine for me, this way we construct a lot new objects!